mixer/pkg/runtime/config/ephemeral.go

```diff
@@ -226,6 +242,160 @@ func (e *Ephemeral) processHandlerConfigs(counters Counters, errs *multierror.Er
 	return handlers
 }
 
+func (e *Ephemeral) processDynamicHandlerConfigs(adapters map[string]*Adapter, counters Counters, errs *multierror.Error) map[string]*HandlerDynamic {
+	handlers := make(map[string]*HandlerDynamic, len(e.adapters))
+
+	for key, resource := range e.entries {
+		isHandler, hasStaticAdapter := e.isHandler(key, resource.Spec)
+		if !isHandler || hasStaticAdapter {
+			// static adapter based handlers are processed elsewhere
+			continue
+		}
+
+		hdl := resource.Spec.(*config.Handler)
+		an1, an2 := canonicalize(hdl.Adapter, constant.AdapterKind, key.Namespace)
+		handlerName := key.String()
+		log.Debugf("Processing incoming handler config: name='%s'\n%s", handlerName, resource.Spec)
+
+		adapter := adapters[an1]
+		if adapter == nil {
+			adapter = adapters[an2]
+		}
+
+		if adapter == nil {
+			appendErr(errs, fmt.Sprintf("handler='%s'.adapter", handlerName),
+				counters.HandlerValidationError, "adapter '%s' not found", hdl.Adapter)
+			continue
+		}
+		// validate if the param is valid
+		bytes, err := validateEncodeBytes(hdl.Params, adapter.ConfigDescSet, getParamsMsgFullName(adapter.PackageName))
+		if err != nil {
+			appendErr(errs, fmt.Sprintf("handler='%s'.params", handlerName),
+				counters.HandlerValidationError, err.Error())
+			continue
+		}
+
+		adapterCfg := &types.Any{
+			TypeUrl: adapter.PackageName + ".Params",
```

Comment:
TypeURL needs an `@type` in the front.
https://github.com/google/protobuf/blob/master/src/google/protobuf/any.proto#L118

Reply:
Good catch Mandar.. The `/` is the most important part of typeUrl.